### ISSUE TITLE
[com_fields] Ability to display field value in different style based on value

### ISSUE THREAD

**Issue by AndySDH** [joomla/joomla-cms#15667]
### Steps to reproduce the issue
Say I have a Radio/List/Checkbox, and I want a color or a certain style associated to each option. I can't do it because no matter the option that is selected and displayed, it always displays the same way.

For example, my radio selection is:
- Green option
- Red option
- Blue option

### Expected result

When the "Green option" is selected, I want to be able to display it with a green color. Or a green background. Whatever. Same for blue and red. I want to be able to customize the appereance of the field in the frontend based on the value displayed.

### Actual result

All values can only be displayed in the same way.

### System information (as much as possible)

Joomla 3.7

### Additional comments

It could be as simple as adding a class to it so that it can then be customized in CSS.

For example, now it displays like this:
`<span class="field-value">Green option</span>`

Add a class:

`<span class="field-value green-option">Green option</span>`

This way I can simply then do

`.field-value.green-option {background-color: green}`

**Comment by Bakual** [comment]
You should be able to do that in a template override. Just create an override for the com_fields -> field JLayout and adjust as needed.
But be aware that having an ID based on an option value can get you duplicate IDs on your page and thus invalid HTML. Maybe better add the value to the class instead.

**Comment by franz-wohlkoenig** [comment]
@AndySDH if Answer by @Bakual solves your Feature Request, please close Issue, thanks.

**Comment by AndySDH** [comment]
Thanks for the reply!

Oh yeah, you're right about adding it to the class being better than having an ID, yes. It was just to convey the example.
Good point about template overrides, but how would I go about getting the option value in order to add it to the class? I believe the $value variable generates the Text (name) and not the value?

Also, I still believe this could be useful to be implemented as part of the native core, after all, it only makes sense to add a class based on the value to allow individual styling, it doesn't do any harm, just possible good for those who need it. What do you think?

**Comment by Quy** [comment]
>Also, I still believe this could be useful to be implemented as part of the native core, after all, it only makes sense to add a class based on the value to allow individual styling,

I assume this would be an issue with punctuation and non-English characters for class names.

**Comment by Bakual** [comment]
> Good point about template overrides, but how would I go about getting the option value in order to add it to the class? I believe the $value variable generates the Text (name) and not the value?

Within the JLayout, `$field->rawvalue` contains the raw value of the field.

**Comment by AndySDH** [comment]
> Within the JLayout, $field->rawvalue contains the raw value of the field.
@Bakual That did the trick :)
Thank you!
Think if you want to consider this as a part of a core edit or not, if not feel free to close the issue 👍

**Comment by AndySDH** [comment]
After testing this a bit, this should not be added to the render field JLayout, because this won't work if multiple values are displayed.
(Note: I of course did an if statement for the field type to make sure to add it only for radio, list and checkboxes)

In case multiple values are selected, the class will show up as "Array", of course, and you won't have a class for each individual value as desired.

So I think this should instead be added where the individual value  gets generated, that means that we should have to end up with something like this:

`<span class="field-value">
   <span class="green-option">Green option</span>
</span>`

And in case of multiple values:

`<span class="field-value">
   <span class="green-option">Green option</span>
   <span class="red-option">Red option</span>
   <span class="blue-option">Blue option</span>
</span>`

But I didn't find where to do this yet.

**Comment by Bakual** [comment]
The actual field value is rendered by the respective formfield plugin. In case of the list one it is here: https://github.com/joomla/joomla-cms/blob/staging/plugins/fields/list/tmpl/list.php#L22-L28
That layout can be overriden as well.

Personally I don't think this should be done in core. It's a quite specific use case which can be solved with overrides and as you pointed out yourself it isn't as easy as it seems and could break in several ways.

I'm closing this issue due to this reasons.

**Comment by AndySDH** [comment]
@Bakual Hey man, sure, understandable.
Thank you very much for pointing me in the right direction, I was indeed able to override those templates, it's a blessing that those plugin templates happen to have been made override-able, since to my knowledge almost no other plugin is. So, thankfully they are, I accomplished exactly what I wanted now.

Take a look: 
![image](https://cloud.githubusercontent.com/assets/25923699/25596184/48e9999a-2ec8-11e7-90b6-eaffe9b3af86.png)
(building a game database)

PS: I also had to remove htmlentities from the echo, otherwise the html tags wouldn't work in the override.

Thanks again
